clang/lib/AST/ExprConstant.cpp

```diff
@@ -8367,7 +8367,13 @@ bool LValueExprEvaluator::VisitVarDecl(const Expr *E, const VarDecl *VD) {
 
     if (auto *FD = Info.CurrentCall->LambdaCaptureFields.lookup(VD)) {
       // Start with 'Result' referring to the complete closure object...
-      Result = *Info.CurrentCall->This;
+      if (auto *MD = cast<CXXMethodDecl>(Info.CurrentCall->Callee);
```

Comment:
Should it be a `dyn_cast` instead?
```suggestion
      if (auto *MD = dyn_cast<CXXMethodDecl>(Info.CurrentCall->Callee);
```

Reply:
Otherwise if `CXXMethodDecl` is expected it doesn't need a non-null check.

Reply:
Because we have established it's a lambda call operator (which can't even be static because we have capture), we know it's a `CXXMethodDecl` (and we don't check MD isn't null)